app/controllers/application_controller.rb

```diff
@@ -25,8 +25,8 @@ def append_info_to_payload(payload)
     payload[:ip] = request.remote_ip
   end
 
-  def analytics
-    @analytics ||= Analytics.new(current_user, request_attributes, ahoy)
+  def analytics(service = Analytics.new(current_user, request))
```

Comment:
I can't either but its doesn't stop the code smelling bad.  It just reads as buggy code as it stands (IMHO).  Happy to not push hard on this and let the change land if you guys agree its ok.